amethyst_renderer/src/pipe/stage.rs

```diff
@@ -37,51 +37,78 @@ impl Stage {
         self.enabled
     }
 
-    /// Get passes of the stage
-    pub fn passes(&self) -> &[Pass] {
-        self.passes.as_slice()
-    }
+    /// Applies all passes within this stage.
+    pub fn apply(&self, encoders: &mut [Encoder], scene: &Scene) {
+        use rayon::prelude::*;
+
+        let mut encs = encoders;
+        self.clear_color.map(|c| self.target.clear_color(&mut encs[0], c));
+        self.clear_depth.map(|d| self.target.clear_depth_stencil(&mut encs[0], d));
+
+        for pass in self.passes.iter() {
+            let mut models = scene.par_chunks_models(4);
 
-    /// Get target of the stage
-    pub fn target(&self) -> &Target {
-        &self.target
+            let enc = {
+                let slice = encs;
+                let (count, left) = slice.split_at_mut(models.len());
+                encs = left;
+                count
+            };
+
+            models.zip(enc)
+                .for_each(move |(models, enc)| for model in models {
```

Comment:
I actually think that all passes should be applied in parallel with only one join at the end. Instead of joining in each stage. That's why I previously moved this code into `Renderer::draw`.